indico/web/client/js/react/components/time/TimePicker.jsx

```diff
@@ -0,0 +1,201 @@
+  const optionValue = option.firstElementChild.dateTime;
+  const keywordValue = Time.fromString(keywordText).toString();
+
+  keywordText = keywordText.toLowerCase().replace(/\s+/, '');
+
+  let optionRank = 0;
+
+  if (keywordText && !optionDisabled) {
+    if (optionValue === keywordValue) {
+      optionRank = Infinity;
+    } else if (optionText.startsWith(keywordText)) {
+      optionRank = 10;
+    }
+  }
+
+  return optionRank;
+}
+
+export default function TimePicker({value, startTime, step, min, max, timeFormat, onChange}) {
+  const [inputValue, setInputValue] = useState(Time.fromString(value, '24h').toString());
+  const [notice, setNotice] = useState('');
+  const valueAsTime = Time.fromString(value, '24h');
+
+  const options = useMemo(() => getOptions(value, startTime, step, min, max, timeFormat), [
+    value,
+    startTime,
+    step,
+    min,
+    max,
+    timeFormat,
+  ]);
+
+  // Auto-clear notice
+  useEffect(() => {
+    if (!notice) {
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      setNotice('');
+    }, 10000);
+
+    return () => clearTimeout(timer);
+  }, [notice]);
+
+  useEffect(
+    () => {
+      const inputTime = Time.fromString(inputValue, 'any');
+      // Using Object.is() as value can also be NaN
+      if (!Object.is(valueAsTime.value, inputTime.value)) {
+        setInputValue(valueAsTime.toFormattedString(timeFormat));
+      }
+    },
+    // The purpose of this hook is to sync the prop changes to the
+    // input value *only* in cases where the prop was change without
+    // first changing the input value. Therefore, the only valid
+    // dependencies are the props. Adding anything from the local state
+    // here would have undesired consequences.
+    //
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [value, timeFormat]
+  );
+
+  const handleChange = ev => {
+    const newValue = ev.target.value;
+    const formattedNewValue = Time.fromString(newValue).toString();
+    setInputValue(newValue);
+    onChange(formattedNewValue === 'Invalid time' ? '' : formattedNewValue);
+  };
+
+  const handleBlur = ev => {
+    // Reformat the value when user leaves the field
+
+    const maybeTime = Time.fromString(ev.target.value);
+    const maybeFixedValue = Number.isNaN(maybeTime.value)
+      ? ''
+      : maybeTime.toFormattedString(timeFormat);
+
+    // Value was already fixed?
+    if (maybeFixedValue === ev.target.value) {
+      return;
+    }
+
+    setInputValue(maybeFixedValue);
+
+    // Announce any changes to screen readers
+
+    if (!maybeFixedValue) {
+      setNotice(
+        Translate.string('The time value was invalid and was cleared', {
+          time: maybeFixedValue,
+        })
+      );
+    } else {
+      setNotice(
+        Translate.string('The time value was automatically formatted as {time}', {
+          time: maybeFixedValue,
+        })
+      );
+    }
+  };
+
+  return (
+    <>
+      <ComboBox
```

Comment:
the `required` prop from the TimePicker needs to be passed to this one so the clear button is not shown in case it's required (I fixed this locally so I'll push a comment for this)